docs/xilinx-zcu102.md

```diff
@@ -0,0 +1,30 @@
+# Notes on using Yoe on the Xilinx ZCU102
+
+[up](README.md)
+
+[BSP Layer documentation](meta-xilinx/meta-xilinx-bsp/README.building.md)
+
+## Building/installing an image
+
+1. `git clone git://github.com/YoeDistro/yoe-distro.git`
+1. `cd yoe-distro`
+1. `. zcu102-zynqmp-envsetup.sh`
+1. `yoe_setup`
+1. add following to conf/local.conf
```

Comment:
Generally we would advise putting these changes in site.conf to be captured in git and shared with the team. local.conf is intended only for settings that are use specific and not checked into git.

https://github.com/YoeDistro/yoe-distro/blob/dunfell/docs/conf-files.md

However, I wonder if these settings can be captured just for these machines (machine conf, etc)?

One thing we try to do with Yoe is keep the system such that you can build for multiple machines in one build system -- this is handy if you are maintaining multiple projects/products.

Reply:
Thanks @cbrake I am adding them to site.conf with machine specific variables, something like this.
diff --git a/conf/site.conf b/conf/site.conf
index f643f48..81d26d5 100644
--- a/conf/site.conf
+++ b/conf/site.conf
@@ -71,6 +71,14 @@ IMAGE_BOOT_FILES_append_sama5d27-som1-ek-sd = "\
 "
 WKS_FILE_sama5d27-som1-ek-sd = "yoe-sdimage.wks"
 
+
+# Xilinx zcu102 specific configurations
+IMAGE_BOOT_FILES_append_zcu102-zynqmp = " boot.bin"
+PREFERRED_PROVIDER_qemu-native_zcu102-zynqmp = "qemu-xilinx-native"
+PREFERRED_PROVIDER_virtual/pmu-firmware_zcu102-zynqmp = "pmu-firmware"
+PREFERRED_PROVIDER_virtual/boot-bin_zcu102-zynqmp = "xilinx-bootbin"
+PREFERRED_PROVIDER_virtual/dtb_zcu102-zynqmp = "device-tree"

It works fine.

Maybe its the inclusion of the Xilinx machine (which also provides QEMU) there is NOTE in the build.

NOTE: Resolving any missing task queue dependencies
NOTE: Multiple providers are available for qemu-native (qemu-native, qemu-xilinx-native)
Consider defining a PREFERRED_PROVIDER entry to match qemu-native
NOTE: Multiple providers are available for runtime nativesdk-qemu (nativesdk-qemu, nativesdk-qemu-xilinx)
Consider defining a PREFERRED_RPROVIDER entry to match nativesdk-qemu


Kindly advice on how to resolve.
Also what to do about these warnings (introduced by Xilinx layers) ?

WARNING: No recipes available for:
  /home/kpoduval/workspace/yoe-zcu/sources/meta-xilinx/meta-xilinx-bsp/recipes-microblaze/gcc/gcc-source_10.%.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-xilinx/meta-xilinx-bsp/recipes-microblaze/newlib/libgloss_3.3.%.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-xilinx/meta-xilinx-bsp/recipes-microblaze/newlib/newlib_3.3.%.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-petalinux/recipes-browser/chromium/chromium-ozone-wayland_%.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-petalinux/recipes-browser/chromium/chromium-x11_%.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-petalinux/recipes-browser/chromium/gn-native_%.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-petalinux/recipes-extended/images/xen-image-minimal.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-petalinux/recipes-extended/xen/xen_git.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-petalinux/recipes-multimedia/gstreamer-1.0/gst-shark_0.2.1.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-petalinux/recipes-support/opencv/opencv_4.3%.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-xilinx/meta-xilinx-standalone/recipes-core/newlib/libgloss_3.1.0.bbappend
  /home/kpoduval/workspace/yoe-zcu/sources/meta-xilinx/meta-xilinx-standalone/recipes-core/newlib/newlib_3.1.0.bbappend

Reply:
I guess those warnings are because of stale recipes from Xilinx layers, I wish there was patch process on the layers themselves :). For the NOTE I think I have it working with defining some defaults before the xilinx specific confs.
+PREFERRED_PROVIDER_qemu-native ?= "qemu-native"
+PREFERRED_PROVIDER_nativesdk-qemu ?= "nativesdk-qemu"

I will put up a new pull request now, thanks.